### DISCUSSION TITLE
How to stop a running node.js server after refreshing code-server

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#2928] [Q&A] [answered]
I was running a node.js express server through the npm scripts panel on the left. If I click the delete button in the terminal, the server stops. If I run the server again and this time I refresh code-server, the terminal is gone but the server is still running. The only way I have been able to stop it is by triggering a container rebuild.

Is there any better way to stop such servers?

Info:
npm package manager: pnpm
platform: railway.app
nodejs: 15

Similar discussions: #2580 (Recover Terminal)

**Comment by jsjoeio** [comment, marked as answer]
cc @bpmct

**Reply by bpmct** [reply]
Would recommend using tmux for something like this so you can always recover those sessions. Yep, this is related to #2580 and I don't think specific to any host (e.g railway).

**Reply by ghost** [reply]
Apparently, tmux is limited to 1 terminal session. Which means you cannot open more than 1 terminal and each time, that 1 will get restored. I guess I'll stick to container rebuild method for now.

**Reply by bpmct** [reply]
Tmux can have multiple windows in a session though, and you can either tile them or toggle between. That should allow you to run many apps and terminals on a single session
